maa-cli/src/dirs.rs

```diff
@@ -394,6 +400,166 @@ fn ensure_name(name: &str) -> &str {
     name
 }
 
+pub trait PathProvider {
+    fn get_path(&self) -> Vec<PathBuf>;
+}
+
+#[derive(ValueEnum, Clone, Debug)]
+pub enum CleanupTarget {
+    CliCache,
+    Avatars,
+    Log,
+    Map,
+}
+
+impl PathProvider for CleanupTarget {
+    fn get_path(&self) -> Vec<PathBuf> {
+        let debug = dirs::log();
+
+        match self {
+            CleanupTarget::CliCache => {
+                vec![cache().to_path_buf()]
+            }
+            CleanupTarget::Avatars => {
+                vec![state().join("cache").join("avatars")]
+            }
+            CleanupTarget::Log => {
+                let log_files = match fs::read_dir(debug) {
+                    Ok(dir) => dir
+                        .filter_map(|entry| {
+                            let entry = entry.ok()?;
+                            let binding = entry.file_name();
+                            let file_name = binding.to_string_lossy();
+                            if file_name.starts_with("20") {
+                                Some(entry.path())
+                            } else {
+                                None
+                            }
+                        })
+                        .collect(),
+                    Err(_) => Vec::new(),
+                };
+                let mut logs = vec![log().join("asst.log"), log().join("asst.bak.log")];
+                logs.extend(log_files);
+                logs
+            }
+            CleanupTarget::Map => {
+                vec![
+                    debug.join("drops"),
+                    debug.join("map"),
+                    debug.join("other"),
+                    debug.join("Roguelike"),
+                ]
+            }
+        }
+    }
+}
+
+pub fn cleanup<T>(targets: &[T]) -> Result<()>
+where
+    T: PathProvider,
+{
+    let targets_to_use: Vec<&dyn PathProvider> = if targets.is_empty() {
+        vec![
+            &CleanupTarget::CliCache,
+            &CleanupTarget::Avatars,
+            &CleanupTarget::Log,
+            &CleanupTarget::Map,
+        ]
+    } else {
+        targets.iter().map(|x| x as &dyn PathProvider).collect()
+    };
+
+    let trash_list: Vec<PathBuf> = targets_to_use
+        .iter()
+        .flat_map(|target| target.get_path())
+        .collect();
+
+    trash_list.iter().enumerate().for_each(|(i, p)| {
+        println!("{}. {}", i + 1, p.display());
+    });
+
+    if !clear_inquiry("files or folders mentioned above?")? {
```

Comment:
We can use `BoolInput` directly here:
```suggestion
    if !BoolInput::new(Some(true), Some("clear files or folders mentioned above")).value()? {
```